knowledge_graphs/performance_monitor.py

```
@@ -0,0 +1,497 @@
+    async def stop_collection(self):
+        """Stop automated metrics collection."""
+        if not self.is_collecting:
+            return
+        
+        self.is_collecting = False
+        if self.collection_task:
+            self.collection_task.cancel()
+            try:
+                await self.collection_task
+            except asyncio.CancelledError:
+                pass
```

Ruff (0.12.2)

Use `contextlib.suppress(asyncio.CancelledError)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(asyncio.CancelledError)`

(SIM105)